pkg/agent/config/config.go

```diff
@@ -38,22 +38,46 @@ const (
 	DefaultPodManifestPath = "pod-manifests"
 )
 
+// Get returns a pointer to a completed Node configuration struct,
+// containing a merging of the local CLI configuration with settings from the server.
+// A call to this will bock until agent configuration is successfully returned by the
+// server.
 func Get(ctx context.Context, agent cmds.Agent, proxy proxy.Proxy) *config.Node {
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+RETRY:
 	for {
 		agentConfig, err := get(ctx, &agent, proxy)
 		if err != nil {
-			logrus.Errorf("Failed to configure agent: %v", err)
-			select {
-			case <-time.After(5 * time.Second):
-				continue
-			case <-ctx.Done():
-				logrus.Fatalf("Interrupted")
+			logrus.Infof("Failed to retrieve agent configuration: %v", err)
+			for range ticker.C {
+				continue RETRY
 			}
 		}
 		return agentConfig
 	}
 }
 
+// KubeProxyDisabled returns a bool indicating whether or not kube-proxy has been disabled in the
+// server configuration. The server may not have a complete view of cluster configuration until
+// after all startup hooks have completed, so a call to this will block until after the server's
```

Comment:
```suggestion
// server configuration. The agent may not have a complete view of cluster configuration until
// after all startup hooks have completed, so a call to this will block until after the agent's
```

A "server" is an "agent" but an "agent" isn't a server and all agents are supposed to run kube-proxy, amirite?

Reply:
Correct, but in this case it's the server's view of the world that we are waiting on, which is why we check readyz on the server we're joining.